test/jdk/com/sun/crypto/provider/Cipher/ChaCha20/unittest/java.base/com/sun/crypto/provider/Poly1305KAT.java

```diff
@@ -0,0 +1,199 @@
+            "4541669a7eaaee61e708dc7cbcc5eb62"));
+        add(new TestData("RFC 7539 A.3 Test Vector #5: If one uses 130-bit partial reduction, does the code handle the case where partially reducedfinal result is not fully reduced?",
+            "0200000000000000000000000000000000000000000000000000000000000000",
+            "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+            "03000000000000000000000000000000"));
+        add(new TestData("RFC 7539 A.3 Test Vector #6: What happens if addition of s overflows modulo 2^128?",
+            "02000000000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+            "02000000000000000000000000000000",
+            "03000000000000000000000000000000"));
+        add(new TestData("RFC 7539 A.3 Test Vector #7: What happens if data limb is all ones and there is carry from lower limb?",
+            "0100000000000000000000000000000000000000000000000000000000000000",
+            "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF0FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF" +
+            "11000000000000000000000000000000",
+             "05000000000000000000000000000000"));
+        add(new TestData("RFC 7539 A.3 Test Vector #8: What happens if final result from polynomial part is exactly 2^130-5?",
+            "0100000000000000000000000000000000000000000000000000000000000000",
+            "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFEFEFEFEFEFEFEFEFEFEFEFEFEFEFE" +
+            "01010101010101010101010101010101",
+            "00000000000000000000000000000000"));
+        add(new TestData("RFC 7539 A.3 Test Vector #9: What happens if final result from polynomial part is exactly 2^130-6?",
+            "0200000000000000000000000000000000000000000000000000000000000000",
+            "FDFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+            "FAFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"));
+        add(new TestData("RFC 7539 A.3 Test Vector #10: What happens if 5*H+L-type reduction produces 131-bit intermediate result?",
+            "0100000000000000040000000000000000000000000000000000000000000000",
+            "E33594D7505E43B900000000000000003394D7505E4379CD0100000000000000" +
+            "0000000000000000000000000000000001000000000000000000000000000000",
+            "14000000000000005500000000000000"));
+        add(new TestData("RFC 7539 A.3 Test Vector #11: What happens if 5*H+L-type reduction produces 131-bit final result?",
+            "0100000000000000040000000000000000000000000000000000000000000000",
+            "E33594D7505E43B900000000000000003394D7505E4379CD0100000000000000" +
+            "00000000000000000000000000000000",
+            "13000000000000000000000000000000"));
+    }};
+
+    public static void main(String args[]) throws Exception {
+        int testsPassed = 0;
+        int testNumber = 0;
+
+        for (TestData test : testList) {
+            System.out.println("*** Test " + ++testNumber + ": " +
+                    test.testName);
+            if (runSingleTest(test)) {
```

Comment:
runSingleTest may need to be called enough number of times for the engineUpdate to be compiled by c2.

Reply:
added a second copy with `@run main/othervm -Xcomp -XX:-TieredCompilation com.sun.crypto.provider.Cipher.ChaCha20.Poly1305UnitTestDriver`